lib/utils.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const scripts = (plugins) => {
+  const s = {};
+
+  if (Array.isArray(plugins)) {
+    plugins.forEach(plugin => {
+      if (typeof plugin.scripts === 'object') {
+        Object.keys(plugin.scripts).forEach(script => {
+          if (!s.hasOwnProperty(script)) {
+            s[script] = plugin.scripts[script];
+          }
+        });
+      }
+    });
+  }
+
+  return s;
+}
+
+const validation = (plugins) => {
+  const v = {};
+
+  if (Array.isArray(plugins)) {
+    plugins.forEach(plugin => {
+      if (typeof plugin.validation === 'object') {
+        Object.keys(plugin.validation).forEach(val => {
+          if (!v.hasOwnProperty(val)) {
+            v[val] = plugin.validation[val];
+          }
+        });
+      }
+    });
+  }
+
+  return v;
+}
+
```

Comment:
These functions are identical except for script/validation. Can they be one function and pass in `key` that would be `'validation'` or `'script'` and return the proper thing from there?

You can create helpers that automatically pass in the correct string if you'd like